### PR TITLE
[Spacing] Button

### PR DESCRIPTION
**Pull request by jonathantneal**
## Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Checklist

- [ ] This PR adds or removes a Storybook story.
- [ ] I have added tests to cover my changes.
- [ ] Regressions are passing and/or failures are documented
- [ ] Changes have been checked in evergreen browsers

### CODE REVIEW

File: packages/web-components/src/components/rux-button/rux-button.scss [hunk 1]
```diff
@@ -1,145 +1,117 @@
...
+    /* references */
+    --disabled-opacity: var(--opacity-40);
+    --size: 0.25rem;
+
+    /* @TODO: Should these be moved into the design tokens library? */
+    /* aliases */
+    --button-font-size: var(--font-body-1-font-size);
+    --button-line-size: var(--line-height-sm);
 
-:host([hidden]) {
```
Comment (markacianfrani) [inline packages/web-components/src/components/rux-button/rux-button.scss:6]: why was this removed?

**Comment by markacianfrani** [comment]
Screen Shot 2022-06-17 at 11 03 15 AM

The secondary borders look off to me. left - png export from figma right - web component. Can you confirm?

File: packages/web-components/src/components/rux-button/rux-button.scss [hunk 2]
```diff
@@ -1,145 +1,117 @@
...
-    width: auto;
-}
+    /* @TODO: Should these be moved into the design tokens library? */
+    /* references */
+    --disabled-opacity: var(--opacity-40);
+    --size: 0.25rem;
+
+    /* @TODO: Should these be moved into the design tokens library? */
+    /* aliases */
+    --button-font-size: var(--font-body-1-font-size);
```
[the remark this answers is not in the document]
Comment (jonathantneal) [reply packages/web-components/src/components/rux-button/rux-button.scss]: These have been removed and put into the global css file in a separate PR merged to the spacing branch.

**Comment by jonathantneal** [comment]
Hey @markacianfrani, thanks for the review. The tokens that were merged into `spacing` have now been applied. I have also addressed the the other issues you discovered. Thank you for spotting those!

File: packages/web-components/src/components/rux-button/rux-button.scss [hunk 3]
```diff
@@ -1,145 +1,117 @@
...
-    text-overflow: ellipsis;
+    cursor: pointer;
+    box-shadow: var(--border-color, transparent) 0px 0px var(--border-width, 0px) inset;
+    display: flex;
     justify-content: center;
-    align-items: center;
     user-select: none;
 
-    rux-icon,
-    ::slotted(rux-icon) {
```
[the remark this answers is not in the document]
Comment (markacianfrani) [reply packages/web-components/src/components/rux-button/rux-button.scss:35]: If its not too much work, I'd say go for it just to avoid having to make a breaking change. I'd say the 'one right way' would actually be the slot if we had to pick one but I don't have strong opinions

**Review by markacianfrani — APPROVED** [review]

File: packages/web-components/src/components/rux-button/rux-button.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonathantneal) [reply packages/web-components/src/components/rux-button/rux-button.scss:35]: This has been added back!
